presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonFileBasedFunctionDefinitionProvider.java

```diff
@@ -28,11 +38,35 @@
     public UdfFunctionSignatureMap getUdfDefinition(String filePath)
     {
         try {
-            return parseJson(Paths.get(filePath), UdfFunctionSignatureMap.class);
+            Map<String, List<JsonBasedUdfFunctionMetadata>> udfSignatureMap = new HashMap<>();
+            for (Path path : getFilesInPath(filePath)) {
+                // Note: Latest updates take effect in case of duplicates.
+                udfSignatureMap.putAll(parseJson(path, UdfFunctionSignatureMap.class).getUDFSignatureMap());
+            }
+            return new UdfFunctionSignatureMap(udfSignatureMap);
         }
         catch (Exception e) {
             log.info("Failed to load function definition for JsonFileBasedFunctionNamespaceManager " + e.getMessage());
         }
         return null;
     }
+
+    // Returns the list containing the file if filePath is a single file or all json files inside filePath if it's a dir
+    private List<Path> getFilesInPath(String filePath) throws IOException
+    {
+        final String jsonFileExtension = ".json";
+        Path path = Paths.get(filePath);
+
+        if (Files.isDirectory(path)) {
+            try (Stream<Path> stream = Files.list(path)) {
+                return stream
+                        .filter(file -> !Files.isDirectory(file))
```

Comment:
can we instead call this recursively to support nested directory structures?  We can use Files.find() to walk the directory tree up to some reasonable depth and also filter to only json files.

Reply:
TBH, I intentionally didn't do this to only get the files at the first level. Is there a desire/need to do this for nested directories?

Reply:
I don't know of an existing need, but I can imagine users might want to create some kind of directory structure to organize their functions, and it's easy enough to support, so why not?

Reply:
I did it this way to avoid including files which are not clearly intended.
Sure. I will update it to include all files recursively.

Reply:
@Ali-P @rschlussel We'd also need to rename `JsonFileBasedFunctionNamespaceManagerConfig .setFunctionDefinitionFile` and update the config description for `json-based-function-manager.path-to-function-definition` but before making these changes, it'd be great to reach agreement w.r.t config interpretation. Assuming we agree on supporting either a single file or a directory for this config, a few subsequent questions to consider are:
1. How do we handle a non-existent path?
2. In case of directory, do we recurse sub directories? 
3. How do we handle an invalid json resource?
4. Extension to 3, how do we handle invalid json resource when enumerating directory? Say, we've an unrelated file mixed into json files, do we ignore and move on or be strict about it and fail?

Reply:
Good points, @ankurpathela! We should reach a consensus on these; some of your suggestions have been addressed in the updated PR.

For reference, the first version of `getFilesInPath` was as follows:
```
// Returns the list containing the file if filePath is a single file or all json files inside filePath if it's a dir
private List<Path> getFilesInPath(String filePath) throws IOException
{
    final String jsonFileExtension = ".json";
    Path path = Paths.get(filePath);

    if (Files.isDirectory(path)) {
        try (Stream<Path> stream = Files.list(path)) {
            return stream
                    .filter(file -> !Files.isDirectory(file))
                    .filter(file -> file.getFileName().toString().toLowerCase(ENGLISH).endsWith(jsonFileExtension))
                    .collect(Collectors.toList());
        }
    }
    else {
        return Collections.singletonList(path);
    }
}
```